### ISSUE TITLE
[nextcloud] Can't get nextcloud to run

### ISSUE THREAD

**Issue by derkd** [Wonderfall/dockerfiles#136]
I'm deploying the nextcloud on a Jelastic docker environment. 
I got this error:
```
[26-Feb-2017 13:37:45] ALERT: [pool www] user has not been defined
[26-Feb-2017 13:37:45] ERROR: failed to post process the configuration
[26-Feb-2017 13:37:45] ERROR: FPM initialization failed
```

I fixed it by adding in /etc/php7.1/php-fpm.conf
```
user = www-data
```

I don't know if this is correct....

The next error is this:

```
Updating permissions...
Updating permissions in /nextcloud...
Permissions in /data are correct.
Permissions in /config are correct.
Permissions in /apps2 are correct.
Permissions in /etc/nginx are correct.
Permissions in /etc/php7.1 are correct.
Updating permissions in /var/log...
Permissions in /var/lib/nginx are correct.
Updating permissions in /tmp...
Permissions in /etc/s6.d are correct.
Done updating permissions.
Nextcloud is not installed - only a limited number of commands are available


[Symfony\Component\Console\Exception\CommandNotFoundException] 
Command "upgrade" is not defined. 


Trying ownCloud upgrade again to work around ownCloud upgrade bug...
Nextcloud is not installed - only a limited number of commands are available


[Symfony\Component\Console\Exception\CommandNotFoundException] 
Command "upgrade" is not defined. 
```

How can I fix this?

**Comment by Wonderfall** [comment]
No need to add the user directive since php is running under GID/UID.
What is you `docker info`?

**Comment by derkd** [comment]
Ok I will undo it.
I can't execute docker info because I'm on a jelastic (www.jelastic.com) environment (which has docker support). Is there something else I can check?

**Comment by MisterCyp** [comment]
EDIT : J'ai supprimé et relancé les conteneurs et ça a fonctionné ...
Salut Wonderfall,

J'ai les mêmes messages d'erreurs : 
```
Attaching to nextcloud
nextcloud       | Updating permissions...
nextcloud       | Updating permissions in /nextcloud...
nextcloud       | Permissions in /data are correct.
nextcloud       | Permissions in /config are correct.
nextcloud       | Permissions in /apps2 are correct.
nextcloud       | Updating permissions in /etc/nginx...
nextcloud       | Updating permissions in /etc/php7...
nextcloud       | Updating permissions in /var/log...
nextcloud       | Updating permissions in /var/lib/nginx...
nextcloud       | Updating permissions in /tmp...
nextcloud       | Updating permissions in /etc/s6.d...
nextcloud       | Done updating permissions.
nextcloud       | Nextcloud is not installed - only a limited number of commands are available
nextcloud       |
nextcloud       |
nextcloud       |   [Symfony\Component\Console\Exception\CommandNotFoundException]
nextcloud       |   Command "upgrade" is not defined.
nextcloud       |
nextcloud       |
nextcloud       | Trying ownCloud upgrade again to work around ownCloud upgrade bug...
nextcloud       | Nextcloud is not installed - only a limited number of commands are available
nextcloud       |
nextcloud       |
nextcloud       |   [Symfony\Component\Console\Exception\CommandNotFoundException]
nextcloud       |   Command "upgrade" is not defined.
```

Voici mon docker info : 

```
Containers: 20
 Running: 12
 Paused: 0
 Stopped: 8
Images: 42
Server Version: 17.03.0-ce
Storage Driver: devicemapper
 Pool Name: docker-8:1-788370-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 24.78 GB
 Data Space Total: 107.4 GB
 Data Space Available: 15.05 GB
 Metadata Space Used: 29.63 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.118 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.90 (2014-09-01)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 977c511eda0925a723debdc94d09459af49d082a
runc version: a01dafd48bc1c7cc12bdb01206f9fea7dd6feb70
init version: 949e6fa
Kernel Version: 3.14.32-xxxx-grs-ipv6-64
Operating System: Debian GNU/Linux 9 (stretch)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.929 GiB
Name: mail
ID: T4NZ:KTVU:ER4S:GURZ:QH67:23ST:GTY5:GF2K:XDXA:3D6B:AYVY:MIE7
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No kernel memory limit support
WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

Merci pour ton aide

**Comment by Hydrog3n** [comment]
@MisterCyp J'ai la même erreur tu as réussi à corriger ?

**Comment by macmeck** [comment]
This should be fixed by 3b97570 if I'm not mistaken... you might want to give it a try.
Unfortunately, daily builds do fail at the moment. We might need to wait until #172 is being sorted out.
